src/index.ts

```diff
@@ -1,21 +1,14 @@
-import { ResolvedConfig, type PluginOption } from "vite"
+import { type PluginOption } from "vite"
 
 import { name as pkgName } from "../package.json"
 import { transform } from "./transform"
 
 export { default as denyImports } from "./deny-imports"
 
-export default (): PluginOption[] => {
-  let root: string
-  let command: ResolvedConfig["command"]
-
+export function envOnlyMacros(): PluginOption[] {
   return [
     {
-      name: "vite-plugin-env-only",
-      configResolved(config) {
-        root = config.root
-        command = config.command
-      },
+      name: "vite-plugin-env-only-macros",
```

Comment:
Nit: Can we shorten this to `vite-env-only-macros` (and the other plugin names too)?

Reply:
From the [Vite Plugin API docs](https://vitejs.dev/guide/api-plugin#conventions)

> Vite Plugins should have a clear name with `vite-plugin-` prefix.

Though its unclear if they mean the package name or the plugin name (I think the latter)

Reply:
Also, `@vitejs/plugin-react` doesn't follow this convention, (neither in package name nor plugin names). Though maybe this convention is for 3rd party plugins where the `@vitejs` namespace wouldn't be present. So that makes me think its a package name convention?

Reply:
`vite-tsconfig-paths` doesn't follow this convention for the package name nor the plugin name either 🤷